dbms/src/DataStreams/WindowBlockInputStream.cpp

```diff
@@ -360,6 +495,142 @@ std::tuple<RowNumber, bool> WindowTransformAction::stepToFrameEnd(const RowNumbe
     return std::make_tuple(frame_end_row, true);
 }
 
+RowNumber WindowTransformAction::stepToFrameStartForRange(const RowNumber & current_row)
+{
+    if (col_type == Window::ColumnType::UnInitialized)
+        // order_column_indices must contain value as tidb should ensure this.
+        col_type = getColumnType(inputAt(current_row)[order_column_indices[0]]);
```

Comment:
Can this be obtained from metadata when compiling WindowFunction?

Reply:
> Can this be obtained from metadata when compiling WindowFunction?

okk, I have changed to this behaviour.